jdaviz/configs/cubeviz/plugins/slice/tests/test_slice.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+
+from jdaviz.configs.cubeviz.plugins.slice.slice import Slice
+
+
+@pytest.mark.filterwarnings('ignore:No observer defined on WCS')
+def test_slice(cubeviz_helper, spectrum1d_cube):
+    app = cubeviz_helper.app
+    sl = Slice(app=app)
+    app.add_data(spectrum1d_cube, 'test')
+    app.add_data_to_viewer("spectrum-viewer", "test")
+    app.add_data_to_viewer("flux-viewer", "test")
+
+    # sample cube only has 2 slices with wavelengths [4.62280007e-07 4.62360028e-07] m
+    assert(sl.slider == 1)
+    cubeviz_helper.select_slice(0)
+    assert(sl.slider == 0)
+
+    with pytest.raises(
+            TypeError,
+            match="slice must be an integer"):
+        cubeviz_helper.select_slice("blah")
+
+    with pytest.raises(
+            ValueError,
+            match="slice must be positive"):
+        cubeviz_helper.select_slice(-5)
+
+    cubeviz_helper.select_wavelength(4.62360028e-07)
```

Comment:
Maybe add a check for invalid wavelength? As shown in screenshot above, it let me enter garbage into the wavelength field.

Reply:
The API call itself requires a float or integer.  The vue logic translates the input string (so that we can handle scientific notation), and _now_ rejects and reverts the input when it fails to convert to a float.